.github/workflows/ci.yml

```diff
@@ -76,11 +76,15 @@ jobs:
         run: cargo llvm-cov --all-features --lcov --output-path lcov.info
 
       - name: Upload Coverage to Codecov
-        uses: codecov/codecov-action@v4
+        uses: Wandalen/wretry.action@v1
         with:
-          token: ${{ secrets.CODECOV_TOKEN }} # not required for public repos
-          files: lcov.info
-          fail_ci_if_error: true
+          action: codecov/codecov-action@v3
+          attempt_limit: 10
```

Comment:
let's have like 3 attempts

Reply:
changed, but 3 attempts likely will not solve the issue 100%, as usually codecov stays down for a couple of seconds/minutes.